### ISSUE TITLE
add rangelimit option

### ISSUE THREAD

**Issue by cipig** [ethereum/go-ethereum#24508]
`eth_getLogs` is causing huge load on the RPC node, using most of the CPU cores

i guess this are the corresponding log entries:
```
Mar 07 05:27:55 electrum1 geth.eth[37875]: WARN [03-07|05:27:55.984] Served eth_getLogs                       conn=70.98.100.xxx:46376   reqid=6      duration=2m26.41494259s  err="context canceled"
Mar 07 05:27:55 electrum1 geth.eth[37875]: WARN [03-07|05:27:55.992] Served eth_getLogs                       conn=70.98.100.xxx:45080   reqid=6      duration=2m58.883381419s err="context canceled"
Mar 07 05:27:56 electrum1 geth.eth[37875]: WARN [03-07|05:27:56.000] Served eth_getLogs                       conn=70.98.100.xxx:45082   reqid=6      duration=2m58.898682623s err="context canceled"
Mar 07 05:27:56 electrum1 geth.eth[37875]: WARN [03-07|05:27:56.004] Served eth_getLogs                       conn=70.98.100.xxx:46380   reqid=6      duration=2m26.313996603s err="context canceled"
```
duration > 2 minutes

can the `--rangelimit` option be added to geth, like it was done in bsc repo?
this is the PR: https://github.com/bnb-chain/bsc/pull/101

**Comment by rjl493456442** [comment]
You can use the following parameters to specify the chain range.

```
fromBlock: QUANTITY|TAG - (optional, default: "latest") Integer block number, or "latest" for the last mined block or "pending", "earliest" for not yet mined transactions.
toBlock: QUANTITY|TAG - (optional, default: "latest") Integer block number, or "latest" for the last mined block or "pending", "earliest" for not yet mined transactions.
```

However, if you want to limit the requests in the server side for protection(for example there are spam requests over the internet to slow down your node), I think you should add rpc endpoint restrictions.

**Comment by cipig** [comment]
Yes, i need to restrict this on server side, since it's a public RPC server. What do you mean with `rpc endpoint restrictions`? I couldn't find any params that i could use to restrict anything (eg the number of cores the server is allowed to use).

**Comment by s1na** [comment]
I'll bring this up for discussion but I feel like it can be easily handled via a proxy/load-balancer in front of geth.

**Comment by holiman** [comment]
Triage discussion: If someone converts the bsc PR to geth, and renamees it to `rpc.rangelimit`, we'd most likely accept that.

**Comment by s1na** [comment]
Is this feature still desired now that we have the accelerated log index? I believe now even whole history queries are possible and will finish in a reasonable time. Also still the general rpc timeouts still apply.

I'd tend towards closing otherwise.

**Comment by MqllR** [comment]
Limiting resource-intensive requests is a beneficial feature for RPC providers. This functionality is also expected to be adopted by Geth forks managing very large databases.

**Comment by MariusVanDerWijden** [comment]
closed by https://github.com/ethereum/go-ethereum/pull/33163, only took 3 years
